### PR TITLE
smart contract calls new page

### PR DESCRIPTION
**Pull request by mihaicalinluca**
#### Did you test the changes locally ?
- [X] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [ ] update documentation that is not relevant anymore
- [X] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/developers/sc-calls.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+A standard sync call performs a direct, synchronous transaction to a contract on the same shard. This type of call is launched with `.sync_call()`.
+
+```rust
+    /// Executes transaction synchronously.
+    ///
+    /// Only works with contracts from the same shard.
+    pub fn sync_call(self) -> <RH::ListReturns as NestedTupleFlatten>::Unpacked
+```
+
```
Comment (BiancaIalangi) [inline docs/developers/sc-calls.md:73]: Would be helpful to say that this is an example and then explain the transaction

File: docs/developers/sc-calls.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+This type of call performs a synchronous call in `readonly` mode, meaning the destination contract's state cannot be altered by this action. This type of call is performed with `.sync_call_readonly()`. 
+
+```rust
+    /// Executes transaction synchronously, in readonly mode (target contract cannot have its state altered).
+    ///
+    /// Only works with contracts from the same shard.
+    pub fn sync_call_readonly(self) -> <RH::ListReturns as NestedTupleFlatten>::Unpacked
+```
+
```
Comment (BiancaIalangi) [inline docs/developers/sc-calls.md:117]: Same here

File: docs/developers/sc-calls.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+### Deploy call
+
+Sync calls can also be used to deploy new contracts. There are two primary methods:
+- Raw deploy (directly deploys a contract with specific bytecode)
+- Deploy from source (deploys a contract using bytecode from an existing contract address)
+
+Syntax-wise, both of these calls are executed using `.sync_call()`, but the transaction setup differs for each type.
+
+For a simple `raw deploy`, initiate a raw deploy transaction as shown below:
```
Comment (BiancaIalangi) [inline docs/developers/sc-calls.md:139]: Better replace `raw deploy` with `raw_deploy`

File: docs/developers/sc-calls.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+## Async calls
+
+Async calls are used when `cross-shard communication` is required, allowing transactions to execute across multiple shards.
+
+### Async call (V1)
+
+The most common type of async call. This type of call can be executed with `.async_call_and_exit()`. As of framework version `0.50.2`, this replaces the deprecated `.async_call()` method.
+
+:::important
+This type of call uses the `async V1` mechanism.
```
Comment (BiancaIalangi) [inline docs/developers/sc-calls.md:178]: Important statements are not recommended to use "this".

File: docs/developers/sc-calls.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    /// - gas
+    /// - a function call, ideally via a proxy.
+    ///
+    /// Value-only promises are not supported.
+    ///
+    /// Optionally, can add:
+    /// - any payment
+    /// - a promise callback, which also needs explicit gas for callback.
+    pub fn register_promise(self)
+```
```
Comment (BiancaIalangi) [inline docs/developers/sc-calls.md:221]: same here

**Review by BiancaIalangi — CHANGES_REQUESTED** [review]

File: docs/developers/sc-calls.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    }
+```
+
+## Async calls
+
+Async calls are used when `cross-shard communication` is required, allowing transactions to execute across multiple shards.
+
+### Async call (V1)
+
+The most common type of async call. This type of call can be executed with `.async_call_and_exit()`. As of framework version `0.50.2`, this replaces the deprecated `.async_call()` method.
```
Comment (andrei-marinica) [inline docs/developers/sc-calls.md:181]: New syntax first, old syntax at the end.

File: docs/developers/sc-calls.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+            .from_source(source)
+            .code_metadata(CodeMetadata::UPGRADEABLE)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call()
+    }
+```
+
+## Async calls
+
+Async calls are used when `cross-shard communication` is required, allowing transactions to execute across multiple shards.
```
Comment (andrei-marinica) [inline docs/developers/sc-calls.md:177]: Point out that they also work in the same shard.

Also talk about them running later and callbacks.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — DISMISSED** [review]

File: docs/developers/developer-reference/sc-to-sc-calls.md [hunk 8]
```diff
@@ -0,0 +1,472 @@
...
+[comment]: # (mx-context-auto)
+
+### Async callbacks
+
+Async calls can optionally register callbacks. The callbacks are called by the system, irrespective of whether the caller completed successfully or failed.
+
+The callback receives the following inputs:
+- The call result, which contains several values:
+    - In case of success:
+        - The status code, 0 in this case, followed by
```
Comment (BiancaIalangi) [inline docs/developers/developer-reference/sc-to-sc-calls.md:84]: it stops suddenly

File: docs/developers/developer-reference/sc-to-sc-calls.md [hunk 9]
```diff
@@ -0,0 +1,472 @@
...
+| Reentrancy issues possible. | Reentrancy not possible. |
+
+
+[comment]: # (mx-context-auto)
+
+### Transfer-execute calls
+
+Transfer-execute calls are basically async calls without callback. You can think of them as a "fire and forget" mechanism.
+
+They come in two flavors:
```
Comment (BiancaIalangi) [inline docs/developers/developer-reference/sc-to-sc-calls.md:46]: it would be easier for the reader to see the flavors if they were highlighted

**Review by BiancaIalangi — CHANGES_REQUESTED** [review]

File: docs/developers/developer-reference/sc-to-sc-calls.md (hunk 9 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-to-sc-calls.md:46]: We're using highlights for code only.
I can make it bold or italic.

File: docs/developers/developer-reference/sc-to-sc-calls.md (hunk 8 above)
Comment (andrei-marinica) [reply docs/developers/developer-reference/sc-to-sc-calls.md]: It was meant to continue on the next line. Rephrased for clarity.

**Review by andrei-marinica — APPROVED** [review]

**Review by BiancaIalangi — APPROVED** [review]

**Pull request merged**